registry/handlers/images.go

```diff
@@ -196,6 +196,7 @@ func (imh *imageManifestHandler) convertSchema2Manifest(schema2Manifest *schema2
 		imh.Errors = append(imh.Errors, v2.ErrorCodeManifestInvalid.WithDetail(err))
 		return nil, err
 	}
+	imh.Digest = digest.FromBytes(manifest.(*schema1.SignedManifest).Canonical)
```

Comment:
Are you sure `Canonical` is correct? This is the of the manifest without any attached signatures.

The HTTP response does have attached signatures, so I would expect we'd want to digest the complete manifest returned by `Payload`.

Of course, it depends whether 1.6 / 1.7 include the signatures when performing this digest calculation. I don't know whether they do or not.